File: twao/villages/TargetVillage.java
package twao.villages;

public class TargetVillage extends Village {
    private int numberOfAttacks;

    public TargetVillage(int x, int y, int numberOfAttacks) {
        super(x, y);
        this.numberOfAttacks = numberOfAttacks;
    }

    public void attack() {
        numberOfAttacks--;
    }

    /**
     * -----------------------------------------------------------
     * getters and setters
     * -----------------------------------------------------------
     */

    public int getNumberOfAttacks() { return numberOfAttacks; }

    public boolean isAssignCompleted() { return numberOfAttacks > 0 ? false : true; }
}

File: src/main/java/io/github/stiv3ns/twactionorganizer/twao/villages/TargetVillage.kt
package io.github.stiv3ns.twactionorganizer.twao.villages

import kotlinx.serialization.Serializable

data class TargetVillage(
        override val x: Int,
        override val y: Int
) : Village {
    override var id: Int? = null

    var numberOfAttacks: Int = 0
        private set

    constructor(x: Int, y: Int, numberOfAttacks: Int) : this(x, y) {
        this.numberOfAttacks = numberOfAttacks
    }

    fun attack() {
        numberOfAttacks--;
    }

    fun isAssignCompleted(): Boolean = (numberOfAttacks <= 0)

    override fun toString() = "$x|$y"
}